docs/4.x/archiving-behavior-specification.md

```diff
@@ -0,0 +1,385 @@
+a jumping off point for manual testing of the whole process.
+
+It will be added to over time, since this process in Matomo is especially complex.
+
+## Core Archiving
+
+_Brief Description of system:_ the entry point for this is the `Piwik\Archive` class which handles both querying
+for archive data, and launching the report aggregation process if archive data isn't found. `Piwik\ArchiveProcessor\Loader`
+is in charge of determining if archiving should be done and then launching the whole process. `PluginsArchiver` is
+the entrypoint for report aggregation.
+
+### General Expected Behavior
+
+API methods use the `Piwik\Archive` class to query for archive data. If we are allowed to archive in the current
+request/process, and cannot find a recent, usable archive, we generate it, by launching the core archiving process.
+
+### Settings
+
+* browser triggered archiving: this setting determines whether log aggregation is allowed to be launched from browser requests.
+  If disabled, this only happens if a segment that is not set to auto-archive is requested (referred to as a custom segment), or
+  if a range period is requested. The setting is stored in two potential places, the `[General] enable_browser_archiving_triggering`
+  INI config setting or the `enableBrowserTriggerArchiving` option value. There are other related INI config settings to disable archiving even if
+  a custom segment or range period is requested: `[General] browser_archiving_disabled_enforce` and `[General] archiving_range_force_on_browser_request`
+  respectively.
+  In the code, the `Rules::isBrowserTriggerEnabled()` checks for this setting.
+* today time to live: this value is the number of seconds before which an archive for today's date is considered valid. If an
+  archive is older than this value, then it is considered outdated. Controlled by a UI setting and the
+  `[General] time_before_today_archive_considered_outdated` INI config option. This is also used as the default value
+  for archives for other periods.
+* `[General] time_before_week_archive_considered_outdated`, `[General time_before_month_archive_considered_outdated]`,
+  `[General] time_before_year_archive_considered_outdated`, `[General] time_before_range_archive_considered_outdated`:
+  specific ttls for different period types. They each default to the 'today time to live' value if not specified
+* custom date ranges to pre-process: there is an INI config setting and some user settings that allow users to specify that
+  certain ranges should be pre-archived. The INI setting is `[General] archiving_custom_ranges`. The user setting is the
+  setting that controls the default period to load in Matomo. These ranges will be processed in core:archive if specified.
+
+### Rules
+
+**When archiving is allowed/disallowed for the current process**
+
+If browser archiving is enabled, any HTTP request or CLI command can initiate archiving. Things are more complicated
+if browser archiving is disabled:
```

Comment:
I don't think the TTL is really about when archiving is allowed/disallowed, it's about when is it necessary/unnecessary. I guess it could be Optimizations. It's described below as well.

Reply:
:+1: